edb/schema/functions.py

```diff
@@ -237,7 +244,7 @@ def as_delete_delta(
         cmd = DeleteParameter(classname=param_name)
 
         if self.type.is_collection() and not self.type.is_polymorphic(schema):
-            param = schema.get(param_name)
+            param: s_types.Type = schema.get(param_name)
```

Comment:
```suggestion
            param = schema.get(param_name, type=Parameter)
```

Also, the annotation of `parent` `cleanup_schema_collection()` is incorrect: it should be `Object`, not `Type`.